src/xml_parser_main_metadata_together.c

```diff
@@ -404,29 +414,23 @@ cr_PkgIterator_parse_next(cr_PkgIterator *iter, GError **err) {
     cr_CbData *cbdata = (cr_CbData*) iter->cbdata;
 
     while (!cr_PkgIterator_is_finished(iter) && g_queue_is_empty(cbdata->finished_pkgs_queue)) {
-        while ((cbdata->in_progress_count_primary <= cbdata->in_progress_count_filelists ||
-                cbdata->in_progress_count_primary <= cbdata->in_progress_count_other) &&
-               !iter->primary_is_done)
+        while (!iter->primary_is_done)
```

Comment:
This has to be: `if (!iter->primary_is_done)`.

Reply:
Ah, whoops.